File: array/leetcode_1.py
# leetcode, Two Sum


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        d_nums = dict()
        for i, k in enumerate(nums):

            if target - k in d_nums:
                return [d_nums[target - k], i]

            d_nums[k] = i

File: array/leetcode_1.kt
package problem_solving

class Solution01 {
    fun twoSum(nums: IntArray, target: Int): IntArray {

        val numMap = mutableMapOf<Int, Int>()

        for (i in nums.indices) {
            numMap[nums[i]] = i
        }

        for (i in nums.indices) {
            if(numMap.contains(target-nums[i])) {
                if(numMap[target-nums[i]] != i) return intArrayOf(i, numMap[target-nums[i]]!!)
            }
        }

        return IntArray(5, {0})


    }
}
